api/app/Generators/CsvGenerator.php

```diff
@@ -15,20 +15,22 @@ public function __construct()
         $this->spreadsheet = new Spreadsheet();
     }
 
-    public function write(string $fileName)
+    public function write(string $fileName, ?string $dir)
     {
-        /** @var \Illuminate\Filesystem\FilesystemManager */
-        $disk = Storage::disk('user_generated');
+        $path = $this->getPath($fileName, $dir);
 
-        $path = $disk->path($fileName);
+        try {
 
-        $writer = new Csv($this->spreadsheet);
-        $writer->setDelimiter(',');
-        $writer->setEnclosure('"');
-        $writer->setLineEnding("\r\n");
-        $writer->setSheetIndex(0);
+            $writer = new Csv($this->spreadsheet);
+            $writer->setDelimiter(',');
+            $writer->setEnclosure('"');
+            $writer->setLineEnding("\r\n");
+            $writer->setSheetIndex(0);
+            $writer->save($path);
 
-        return $writer->save($path);
+        } catch (\Exception $e) {
+            Log::error('Error saving csv: '.$fileName.' '.$e->getMessage());
```

Comment:
I feel like we should have an error notification to send when something goes wrong :thinking:

Reply:
can we bubble this up to say "There was a problem on our end. <filename> failed to download. If you continue to receive this error, please get in touch with our support team." ?

Reply:
okay, we now have an error notification. The message could probably be improved :sweat_smile: 

[add error notification when job fails](https://github.com/GCTC-NTGC/gc-digital-talent/pull/10992/commits/3cee28d0d66a37cac00cf45ec69957a21861abaf)